src/Microsoft.DotNet.XHarness.Common/Execution/ProcessManager.cs

```diff
@@ -286,16 +262,17 @@ private static async Task KillTreeAsyncInternal(int pid, Action<int, int> kill,
                     }
 
                     string? a = null, b = null;
-                    currentEnvironment?.TryGetValue(key!, out a);
-                    processEnvironment?.TryGetValue(key!, out b);
+                    currentEnvironment?.TryGetValue(key, out a);
+                    processEnvironment?.TryGetValue(key, out b);
                     if (a != b)
                     {
                         sb.Append($"{key}={StringUtils.Quote(b)} ");
                     }
                 }
             }
+
             sb.Append($"{StringUtils.Quote(process.StartInfo.FileName)} {process.StartInfo.Arguments}");
-            log.WriteLine(sb);
+            log.WriteLine(sb.ToString());
```

Comment:
this doesn't change the behavior and at least on my VS would have a "you don't need to do this" squiggle.   Same for many of your ToString()s

Reply:
I removed the `StringBuilder` overloads because they are only used here and all implementations of `ILog` had to implement this (in the same way)